live-build/build.gradle

```diff
@@ -52,6 +54,51 @@ def runTypesAndSuffixes = [[upgradeArtifactsRunType, "UpgradeArtifacts"],
                            [vmArtifactsRunType, "VmArtifacts"],
                            [allRunType, ""]]
 
+
+
+task generateHotfixMetaData(type: Exec) {
```

Comment:
A separate gradle task provides:

1. Checking parameters before running other build steps, failing the build early rather than spending time to only then fail later.
2. Allows shorter and easier dev-test cycles via gradle.
3. `generateHotfixMetadata` can take a while to run and so is better to use gradle, which support not re-running tasks that don't have changed inputs.

The intent is to make the hotfix metadata an input of the build rather than something the build generates and uses.

Reply:
> The intent is to make the hotfix metadata an input of the build rather than something the build generates and uses.

Hm, this has me thinking.. would it be better, from a build architecture stand-point, to move the script that builds the metadata file into the devops-gate.. such that we could call it via a new Jenkins job, and have it store it's output in S3.. such that appliance-build would just consume the file from S3? rather than having appliance-build run the script, and generate it for each build?

Reply:
Further, we likely don't really want this script to be versioned alongside the product.. and since appliance-build is currently versioned based on the product version, that already makes me think that having this in the appliance-build repository is not the ideal location..

Reply:
We also seem to run appliance build for multiple variants as separate tasks, which means the hotfix metadata file is generated multiple times. Although the variants are done in parallel, it adds a few minutes to each build.

Reply:
oh, yea.. there's that, too.

how do you feel about potentially moving this script into a separate job, like I was thinking?